flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/autoscaler/KubernetesScalingRealizerTest.java

```diff
@@ -48,8 +48,8 @@ public void testApplyOverrides() {
                                 .get(PipelineOptions.PARALLELISM_OVERRIDES.key()))
                 .satisfiesAnyOf(
                         // Currently no enforced order inside the overrides string
-                        overrides -> assertThat(overrides).isEqualTo("a:1,b:2"),
-                        overrides -> assertThat(overrides).isEqualTo("b:2,a:1"));
+                        overrides -> assertThat(overrides).isEqualTo("{a: '1', b: '2'}"),
+                        overrides -> assertThat(overrides).isEqualTo("{b: '2', a: '1'}"));
```

Comment:
What is going on here? Did the format change somehow? I am a bit confused

Reply:
I was trying to give an explanation in the PR desc :) But the main thing is that now [`GlobalConfiguration` has a `standardYaml` field](https://github.com/apache/flink/blob/a2d42b745b1b337bc92d4cd780b7126b4f2f7121/flink-core/src/main/java/org/apache/flink/configuration/GlobalConfiguration.java#L73), which affects [`ConfigurationUtils#convertValue`](https://github.com/apache/flink/blob/a2d42b745b1b337bc92d4cd780b7126b4f2f7121/flink-core/src/main/java/org/apache/flink/configuration/ConfigurationUtils.java#L524), which is used to read back the `overrides` result in this test case. If `standardYaml` would be `false` the result string would be in the old format, but by default it is `true` and here we have a dummy `Configuration` obj.

Reply:
we may want to disable this by default as this can cause backward compatibility issues with the autoscaler stored override values

Reply:
Just as a side note, someone on Flink Slack mentioned that they cannot use the configuration `pipeline.serialization-config` with the operator because that would require the new YAML parser.

I think it would make sense to add support for the new `config.yaml` file for Flink resources and possibly for the operator itself in the future.

Reply:
Agree with @mateczagany, I started already to play with this. Will create a ticket for that soon.